app/models/miq_server/role_management.rb

```diff
@@ -81,14 +81,14 @@ def set_role_activation(active, *roles)
 
     ids = roles == ["*"] ? server_roles.pluck(:id) : ServerRole.where(:name => roles).pluck(:id)
     assigned_server_roles.where(:server_role_id => ids).each do |a|
-      next if a.database_owner?
+      next if a.name == "database_owner"
```

Comment:
I agree that this method was strange on the AssignedServerRole object, but I don't really like spreading around string comparisons with the role name.

I think I would prefer to keep the knowledge about how to find that particular role in ServerRole and compare the classes here. What do you think?